README.md

```diff
@@ -0,0 +1,229 @@
+		 *
+		 * Examples:
+		 * --------------------
+		 * SimpleEvent event = new SimpleEvent();
+		 * onSimple.emit(event);
+		 * --------------------
+		 */
+	}
+}
+
+public bool onFooSimple(Foo foo, SimpleEvent event)
+{
+	// logic here
+	// return true or false to stop or not propagating the event respectively
+	return true;
+}
+
+public void main()
+{
+	Foo foo = new Foo();
+
+	foo.onSimple.connect((Foo foo, SimpleEvent event) {
+		// logic here
+		// return true or false to stop or not propagating the event respectively
+		return true;
+	});
+
+	// same as
+	import std.functional : toDelegate;
+	foo.onSimple.connect(toDelegate(&onFooSimple));
+
+	// calls onFooSimple
+	foo.simple();
+}
+```
+
+### A more realistic example
+Lets imagine an use case where you have mouse events
+```d
+module mouseevent;
+import deventsystem.event;
+
+@EventType("MouseEvent")
+class MouseEvent : Event
+{
+	mixin basicEventType!MouseEvent;
+
+	protected this() {}
+}
+
+@EventType("MouseButtonEvent")
+class MouseButtonEvent : MouseEvent
+{
+	mixin basicEventType!MouseEvent;
+
+	protected this(uint keyval)
+	{
+		this.keyval = keyval;
+	}
+
+	const uint keyval;
+}
+
+@EventType("MouseButtonPressedEvent")
+class MouseButtonPressedEvent : MouseButtonEvent
+{
+	mixin basicEventType!MouseButtonPressedEvent;
+
+	public this(uint keyval)
+	{
+		super(keyval);
+	}
+}
+
+@EventType("MouseButtonReleasedEvent")
+class MouseButtonReleasedEvent : MouseButtonEvent
+{
+	mixin basicEventType!MouseButtonReleasedEvent;
+
+	public this(uint keyval)
+	{
+		super(keyval);
+	}
+}
+```
+Lets imagine you have program with windows which capture mouse events and now
+you want to add functionality to that
+```d
+module window
+import mouseevent;
+import deventsystem.event;
+import std.typecons : scoped;
+
+class Window
+{
+	mixin genCallback!(Window, MouseButtonPressedEvent) onMouseButtonPressed;
+
+	// you can create a callback with some of the variables in an Event!
+	// be aware the variable's name MUST BE equal to the names in the event
+	// class
+	mixin genCallback!(Window, MouseButtonReleasedEvent, int, "keyval") onMouseButtonReleased;
+
+public:
+	this()
+	{
+		// you can add some default behaviour
+		onMouseButtonPressed.connect((Window, MouseButtonPressedEvent event) {
+			// logic here
+			return true;
+		});
+
+		onMouseButtonReleased.connect((Window, int keyval) {
+			// logic here
+			return true;
+		});
+	}
+
+	void mouseButtonPress(uint keyval)
+	{
+		// some logic ...
+		// emit the signal
+		onMouseButtonPressed.emit(scoped!MouseButtonPressedEvent(keyval));
+	}
+
+	void mouseButtonRelease()
+	{
+		// some logic ...
+		// emit the signal
+		onMouseButtonReleased.emit(keyval);
+	}
+
+protected:
+	void onEvent(Event event)
+	{
+		import deventsystem.eventdispacher : EventDispacher;
+		auto ed = scoped!EventDispacher(event);
+
+		ed.dispach!MouseButtonPressedEvent(&onMouseButtonPressed.dispach);
+		ed.dispach!MouseButtonReleasedEvent(&onMouseButtonReleased.dispach);
+	}
+}
```

Comment:
You forgot the backticks at the end of the file \`\`\` . Don't forget to leave a newline at the end to comply with POSIX standard.

Reply:
done